Dup selector string when extracting in case it's frozen

Tonkpils opened the pull request:
Looking forward to Ruby 3 where strings will be immutable by default and Ruby 2 with `# frozen_string_literal: true`, this prevents raising modification of frozen string errors.

This is pointing to the `1-0-stable` as it has been fixed on master 😄 

Fixes https://github.com/rails/rails-dom-testing/issues/69

kaspth commented on lib/rails/dom/testing/assertions/selector_assertions/html_selector.rb:
```diff
@@ -74,8 +74,7 @@ def extract_selector
 
-    context.substitute!(selector, @values)
-    selector
+    @selector = context.substitute!(selector, @values)
```
The `!` indicates `substitute!` mutates the argument, so why don't we put the `dup` in here?

Tonkpils commented:
> Thanks! Can you file this against master with addressed review, then I'll handle the backports and release :)

@kaspth thanks for reviewing this! I made the changes in a similar fashion to https://github.com/rails/rails-dom-testing/commit/b810e9b59bd6da1d0e3cca001b4fd2d6f0cf418b as that is the fix that was made on master and it is why I did not point the fix there. Would you still like me address your review for the master branch or how would you like me to proceed?

kaspth commented:
Ah, it's been so long since I last looked at this that I didn't know it was fixed on master. Soldiering on here is fine 👍
